beesdoo_worker_status/models/cooperative_status.py

```diff
@@ -350,16 +350,19 @@ def _get_irregular_worker_domain(self, today):
         ]
 
     def _change_irregular_counter(self):
-        if self.sr > 0:
-            if self.sr >= 1:
-                self.is_penalised_irregular = False
-            self.sr -= 1
-        elif self.alert_start_time or self.is_penalised_irregular:
-            self.sr -= 1
+        if self.env["ir.config_parameter"].sudo().get_param("irregular_penalty"):
+            if self.sr > 0:
+                if self.sr >= 1:
+                    self.is_penalised_irregular = False
+                self.sr -= 1
+            elif self.alert_start_time or self.is_penalised_irregular:
+                self.sr -= 1
+            else:
+                self.sr -= 2
+                # Until sr is 1 or above, don't do another penalty.
+                self.is_penalised_irregular = True
```

Comment:
This seems more readable to me :

```suggestion
        irregular_penalty =  self.env["ir.config_parameter"].sudo().get_param("irregular_penalty")
        if irregular_penalty:
            if self.sr >= 1:
                self.sr -= 1
                self.is_penalised_irregular = False    
            elif self.sr == 0:
                self.sr -= 1
            elif self.alert_start_time or self.is_penalised_irregular:
                self.sr -= 1
            else:
                self.sr -= 2
                # Until sr is 1 or above, don't do another penalty.
                self.is_penalised_irregular = True
```

Reply:
Or 

```py
        irregular_penalty =  self.env["ir.config_parameter"].sudo().get_param("irregular_penalty")
        if irregular_penalty:
            if self.sr < 0 and not (self.alert_start_time or self.is_penalised_irregular):
                self.sr -= 2
                # Until sr is 1 or above, don't do another penalty.
                self.is_penalised_irregular = True
            else:
                if self.sr >= 1:
                    self.is_penalised_irregular = False
                self.sr -= 1
```

Reply:
Fixed (with first version, this method is complex, I prefer avoiding risks to change it too much)